baggage_setter.go

```diff
@@ -44,7 +44,7 @@ func newBaggageSetter(restrictionManager baggage.RestrictionManager, metrics *Me
 func (s *baggageSetter) setBaggage(span *Span, key, value string) {
 	var truncated bool
 	var prevItem string
-	restriction := s.restrictionManager.GetRestriction(key)
+	restriction := s.restrictionManager.GetRestriction(span.tracer.serviceName, key)
```

Comment:
if we're moving in this direction, you may want to do `span.serviceName()` which currently would resolve to `self.tracer.serviceName`, but in the future may look it up from tags or whatever.